pandas/_libs/parsers.pyx

```diff
@@ -1267,6 +1267,8 @@ cdef class TextReader:
             return self._string_convert(i, start, end, na_filter,
                                         na_hashset)
         elif is_categorical_dtype(dtype):
+            # TODO: I suspect that this could be optimized when dtype
```

Comment:
I haven't spent any time optimizing this. It could presumably be made faster when we know the categories ahead of time.

Reply:
I agree.  There should be a fastpath for this (or at least implement a different method to extract them).